extensions/auth/portability-auth-microsoft/src/main/java/org/dataportabilityproject/auth/microsoft/ParamStringBuilder.java

```diff
@@ -0,0 +1,52 @@
+package org.dataportabilityproject.auth.microsoft;
+
```

Comment:
I agree we should avoid reinventing the wheel here but my concern would be bringing in a library for one class. This may go back to the issue about modularity. In my mental model, the Microsoft extensions may be running in a process by themselves (e.g. transfer worker, customized auth system, etc) so they may not have libraries used by other extensions available to them.

Is the Apache library something we need to have loaded, in which case we should definitely use it, or is it something that is optional?

Reply:
Agreed, if we don't use the library for anything else, we may just want to roll our own solution if it's a few cases. 

Since it's very general, should this exist in a utilities package and we can swap it out if need be later? 

Also, Guava doesn't seem to be getting this functionality soon. Another option is OKHttp https://github.com/google/guava/issues/1005

Reply:
I'd be happy to move this up to an SPI package used by transfer, auth, and cloud extensions. 

I had a question about OKHttp. I used that in the MS extension out of habit since I've used the library in the past with success. I quickly tried the Google HTTP library used by other extensions but ran into a few issues which were probably the result of me not spending the time figuring out how to use it. In any event, maybe we want to switch over to the Google library? I figured it would be good to first get things working since extensions used by different modules will become easier to detect once the project is moved over to the new structure.

Reply:
Agreed on rolling our own solution and moving this to a shared utils directory later. A suggestion for later use to make this take a Map<String, List<String>>. That way it mimics the apache library and would avoid the state keeping transitions.

Reply:
Quick Q, where would that method go? Here's how I'm using the builder:

https://gist.github.com/jimmarino/ec864ae4570d170ac7d83ed1d2b194ae